backend/learnings/views.py

```diff
@@ -164,3 +169,20 @@ class UserDetails(generics.RetrieveAPIView):
     queryset = EUser.objects.all()
     serializer_class = EUserSerializer
     lookup_field = "username"
+
+
+class EditUserDetails(generics.UpdateAPIView):
+    queryset = EUser.objects.all()
+    serializer_class = EditUserSerializer
+    lookup_field = "id"
+
+    def update(self, request, *args, **kwargs):
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=True)
+        if serializer.is_valid():
+            user = User.objects.get(username=instance.username)
+            user.set_password(serializer.validated_data["password"])
+            user.username = serializer.validated_data["username"]
+            user.save()
+            serializer.save()
```

Comment:
instead of using if
```
serializer.is_valid(raise_exception=True)
user = User.objects....
user.set_password...
```
this way it will raise an error and the user will know that the request failed

Note:
The suggested code is same as this
```
if serializer.is_valid():
    ... do something
else:
    raise validation error
```